File: 算法/代码随想录算法PDF/00调试源代码/559.n-叉树的最大深度.cpp
/*
 * @lc app=leetcode.cn id=559 lang=cpp
 *
 * [559] N 叉树的最大深度
 */

// @lc code=start
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    int maxDepth(Node* root) {
        queue<Node*> que;
        int depth = 0;
        if (root) que.push(root);
        while (!que.empty()) {
            depth++;
            int size = que.size();
            for (int i = 0; i < size; i++) {
                Node* node = que.front();
                que.pop();
                for (auto& ch : node->children) {
                    if (ch) que.push(ch);
                }
            }
        }
        return depth;
    }
};
// @lc code=end

File: 算法/代码随想录算法PDF/00调试源代码/559.n-叉树的最大深度.py
#
# @lc app=leetcode.cn id=559 lang=python3
#
# [559] N 叉树的最大深度
#

# @lc code=start
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def maxDepth(self, root: 'Node') -> int:
        que = deque([])
        depth = 0
        if root:
            que.append(root)
        while que:
            depth += 1
            size = len(que)
            while size:
                node = que.popleft()
                for ch in node.children:
                    if ch:
                        que.append(ch)
                size -= 1
        return depth
# @lc code=end

